tests/unit_tests_Sampler_functionality.bash

```diff
@@ -69,6 +71,38 @@ function Clean_Tests_Environment_For_Following_Test__Sampler-create-input-file()
     rm -r "${HYBRID_output_directory}"
 }
 
+function Make_Test_Preliminary_Operations__Sampler-create-input-file-with-FIST()
+{
+    __static__Do_Preliminary_Sampler_Setup_Operations
+    HYBRID_module[Sampler]='FIST'
+    touch "${HYBRID_fist_module[Particle_file]}"
+    touch "${HYBRID_fist_module[Decays_file]}"
+    Perform_Sanity_Checks_On_Provided_Input_And_Define_Auxiliary_Global_Variables
+
+}
+
+function Unit_Test__Sampler-create-input-file-with-FIST()
+{
+    mkdir -p "${HYBRID_software_output_directory[Hydro]}"
+    touch "${HYBRID_software_output_directory[Hydro]}/freezeout.dat"
+    Call_Codebase_Function_In_Subshell Prepare_Software_Input_File_Sampler
+    if [[ $? -ne 0 ]]; then
+        Print_Error 'Preparation of input unexpectedly failed.'
+        return 1
+    elif [[ ! -f "${HYBRID_software_configuration_file[Sampler]}" ]]; then
+        Print_Error 'The output directory and/or software input file were not properly created.'
+        return 1
+    fi
+}
+
+function Clean_Tests_Environment_For_Following_Test__Sampler-create-input-file-with-FIST()
+{
+    HYBRID_module[Sampler]='SMASH'
```

Comment:
This is suspicious here, as tests are run in sub-shells and we never deal with variables in clean-up. 🤔 I would simply set this variable in the beginning of unit tests as you do in the preliminary operations either to `SMASH` or to `FIST`.